### PR TITLE
Migrate automatic documentation from Travis to GH Actions

### LINKED ISSUE #293: Set up automatic documentation in Github actions

### PR DESCRIPTION
**Pull request by SJaffa**
First try at migrating documentation from Travis to GH Actions

### CODE REVIEW

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+on:
+  push:
+    branches: [ development ]
+  pull_request:
+    branches: [ development ]
+  
+jobs:
+  make-documentation:
+    runs-on: ubuntu-latest
```
Comment (tkoskela) [inline .github/workflows/documentation.yml]: I would fix the version of ubuntu just to be on the safe side
```suggestion
    runs-on: ubuntu-20.04
```

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v2
+    
+      - name: Install dependencies
+        run: sudo apt install libfftw3-dev libtiff5-dev libboost-all-dev libeigen3-dev libyaml-cpp-dev libcfitsio-dev casacore-dev libllvm7 
+    
+      - name: before script #extra dependencies? trusty?
+        run: |
+          curl https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+          echo "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty-7 main" | sudo tee -a /etc/apt/sources.list
```
Comment (dpshelio) [inline .github/workflows/linting.yml]: ubuntu-latest is 20.04 [according to GH's vritual envs](https://github.com/actions/virtual-environments#available-environments)
```suggestion
          echo "deb http://apt.llvm.org/focal/ llvm-toolchain-focal main" | sudo tee -a /etc/apt/sources.list
```
you can find llvm details in [their apt page](https://apt.llvm.org/)

File: .github/workflows/linting.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+          sudo apt-get update -y -qq 
+          sudo apt-get install clang-format-7
```
Comment (dpshelio) [inline .github/workflows/linting.yml]:
```suggestion
          sudo apt-get install clang-format
```

File: .github/workflows/linting.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          mkdir build
+          cd build
+          cmake .. \
+          -Ddompi=OFF \
+          -Dopenmp=OFF \
+          -Ddocs=OFF 
+
+      - name: Build #?
+        run: | 
+          find ../ -regex '.*\.\(cc\|h\)' -not -iname '*.in.h' | xargs -I{} -P 10 clang-format-7 -i -style=file {}; git diff
```
Comment (dpshelio) [inline .github/workflows/linting.yml]:
```suggestion
          find ../ -regex '.*\.\(cc\|h\)' -not -iname '*.in.h' | xargs -I{} -P 10 clang-format -i -style=file {}; git diff
```

**Review by dpshelio — CHANGES_REQUESTED** [review]
clang-format 7 is not available for latest ubuntu version, we need to check whether it works with the new one.

**Review by tkoskela — CHANGES_REQUESTED** [review]

File: .github/workflows/linting.yml (hunk 2 above)
Comment (SJaffa) [reply .github/workflows/linting.yml]: This is the linting action which should have been on a separate branch, I've made the requested changes on the correct branch and removed this file from this branch.

File: .github/workflows/linting.yml (hunk 3 above)
Comment (SJaffa) [reply .github/workflows/linting.yml]: See previous comment on linting

File: .github/workflows/linting.yml (hunk 4 above)
Comment (SJaffa) [reply .github/workflows/linting.yml]: See previous comment on linting

**Review by tkoskela — APPROVED** [review]

**Review by dpshelio — APPROVED** [review]

**Pull request merged**